packages/react-components/src/components/chart/store/dataStreamMinMaxStore.ts

```diff
@@ -0,0 +1,32 @@
+import { StateCreator } from 'zustand';
+
+export type MinMaxMap = { [key in string]: number };
+
+export interface MinMaxData {
+  dataStreamMaxes: MinMaxMap;
+  dataStreamMins: MinMaxMap;
+}
+
+export interface MinMaxState extends MinMaxData {
+  setMax: (datastreamId: string, maxValue: number) => void;
+  setMin: (datastreamId: string, minValue: number) => void;
+}
+
+export const createMinMaxSlice: StateCreator<MinMaxState> = (set) => ({
+  dataStreamMaxes: {},
+  dataStreamMins: {},
+  setMax: (datastreamId, value) =>
+    set((state) => ({
+      dataStreamMaxes: {
```

Comment:
what happens when we remove datastreams? do they stick around in the store?

Reply:
It would be possible to aggregate all of the values at once from the callback, and then overwrite all the mins/ maxes so that we never have a scenario where we leave orphaned datastream mins/maxes

Reply:
thats genius. Done!